addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,198 @@
+                            break;
+                        }
+                        nhcComm = null;
+                    }
+                    if (!nhcComm.communicationActive()) {
+                        IOException e = new IOException("Niko Home Control: communication socket not open");
+                        throw e;
+                    }
+                    thing.getConfiguration().put(CONFIG_HOST_NAME, nhcComm.getAddr().getHostAddress());
+                    thing.getConfiguration().put(CONFIG_PORT, port);
+                    HashMap<String, String> properties = new HashMap<String, String>();
+                    properties.put("IP Address", nhcComm.getAddr().getHostAddress());
+                    properties.put("Port", Integer.toString(port));
+                    properties.put("swversion", nhcComm.getSwversion());
+                    properties.put("api", nhcComm.getApi());
+                    properties.put("time", nhcComm.getTime());
+                    properties.put("language", nhcComm.getLanguage());
+                    properties.put("currency", nhcComm.getCurrency());
+                    properties.put("units", nhcComm.getUnits());
+                    properties.put("DST", nhcComm.getDst());
+                    properties.put("TZ", nhcComm.getTz());
+                    properties.put("lastenergyerase", nhcComm.getLastenergyerase());
+                    properties.put("lastconfig", nhcComm.getLastconfig());
+                    thing.setProperties(properties);
+                    updateStatus(ThingStatus.ONLINE);
+
+                    nhcDiscovery.start(bundleContext);
+                    nhcDiscovery.discoverDevices();
+
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                            "Niko Home Control: error initializing bridge handler");
+                }
+            }
+        });
+
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(int refreshInterval) {
+
+        if (refreshTimer != null) {
+            refreshTimer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if (refreshInterval == 0) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Niko Home Control: restart bridge connection every {} min", refreshInterval);
+        refreshTimer = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Niko Home Control: restart communication at scheduled time");
+                updateStatus(ThingStatus.OFFLINE);
+                nhcComm.stopCommunication(true);
+                thing.setProperty("IP Address", nhcComm.getAddr().getHostAddress());
```

Comment:
Separated the whole properties block out in a separate method, so no duplication anymore.